DBM-Naxx/MilitaryQuarter/Gothik.lua

```diff
@@ -105,6 +110,20 @@ end
 --	end
 -- end
 
+function mod:TeleportLoop()
+	timerTeleport:Start()
+	warnTeleportSoon:Schedule(17)
+	self:ScheduleMethod(20, "TeleportLoop")
+end
+
+function mod:UNIT_HEALTH(uId)
+	if self:GetUnitCreatureId(uId) == 16060 and UnitHealth(uId) / UnitHealthMax(uId) <= 0.30 then
```

Comment:
TL;DR: Use a custom `hp30` sync event for visibility range/lag handling.

Ok, this seems like a good way to check for 30% hp. However, it will only work _for one user_. Suppose someone is ported at 30.01% and doesn't see the `UNIT_HEALTH` event in combat log due to server implementation.

Now, there is a syncing mechanism for that. Change it the following way:
- everyone checks for 30% hp in `mod:UNIT_HEALTH(uId)`, let's call this custom event `hp30`
- instead of handling it immediately, `self:SendSync("hp30")` is done
- then, in `mod:OnSync(event)` the handling happens (`Unschedule/Stop`)
- DBM automatically de-duplicates sync events (so you won't see it 5 times)

So in effect, multiple people track the boss HP. Least lag player with DBM will "win" the sync. Raid assistant/leader's DBM will handle it & stop all relevant timers.

Useful links:
- Sending sync example: https://github.com/SiarkowyForks/dbm-wotlk5man-lfr/blob/a3cdc22390d3a1404329ed20722a931c5b710355/DBM-Naxx/FrostwyrmLair/Sapphiron.lua#L60
- Handling sync example: https://github.com/SiarkowyForks/dbm-wotlk5man-lfr/blob/a3cdc22390d3a1404329ed20722a931c5b710355/DBM-Naxx/FrostwyrmLair/Sapphiron.lua#L67
- How it's used for encounters: https://github.com/SiarkowyForks/dbm-wotlk5man-lfr/search?p=2&q=sync

Reply:
That's not how this particular teleport works, but i can see the issue if not everyone is currently targeting the boss, which send sync function should swiftly fix :)